web/src/message_events.js

```diff
@@ -225,7 +226,10 @@ export function update_messages(events) {
             msg.raw_content = event.content;
         }
 
-        unread.update_message_for_mention(msg, any_message_content_edited);
+        if (unread.update_message_for_mention(msg, any_message_content_edited)) {
+            const topic_key = recent_topics_util.get_topic_key(msg.stream_id, msg.topic);
+            recent_topics_ui.inplace_rerender(topic_key);
+        }
```

Comment:
The function is also called from `process_unread_message`, so it's likely a bug to have move these lines to only one of the callers. Please check for this sort of thing with `git grep` when doing any refactoring -- it's really important to avoid introducing bugs.

Reply:
I have checked `process_unread_message`, but noticed that `update_message_for_mention` is only calling `recent_topics_ui.inplace_rerender(topic_key)` when `content_edited` is `true`. By default, `content_edited` is `false` if the second argument is not passed to `update_message_for_mention`. Since `process_unread_message` is not passing a second argument, `content_edited` is `false` in this case and the `inplace_rerender` function won't be called. Therefore, I did not add these lines to the `process_unread_message` call.

Reply:
OK, this makes sense -- please provide this sort of explanation in the commit message in the future, since it's very important to verifying your changes are correct.